R/ts_cluster.R

```diff
@@ -111,10 +113,26 @@ exp_ts_cluster <- function(df, time, value, category, time_unit = "day", fun.agg
       # Drop columns (represents category) that has more NAs than max_category_na_ratio, considering them to have not enough data.
       df <- df %>% dplyr::select_if(function(x){sum(is.na(x))/length(x) < max_category_na_ratio})
       if (length(colnames(df)) <= centers) {
-        stop("EXP-ANA-2 :: [] :: There is not enough data left after removing high NA ratio data.")
+        if (stop_for_no_data) {
+          stop("EXP-ANA-2 :: [] :: There is not enough data left after removing high NA ratio data.")
+        }
+        # For Analytics View, keep going to show at least the diagnostic chart to show where NAs are.
+        model <- list()
+        # Pass original data.
+        # - So that we can generate diagnostic chart about where NAs are.
+        attr(model, "aggregated_data") <- df_summarised
+        attr(model, "error") <- "There is not enough data left after removing high NA ratio data."
+        class(model) <- "PartitionalTSClusters_exploratory"
+        return(model)
       }
       # Fill NAs in time series
       df <- df %>% dplyr::mutate(across(-time, ~fill_ts_na(.x, time, type = na_fill_type, val = na_fill_value)))
+
+      if (!is.null(roll_mean_window) && roll_mean_window > 1) {
+        df <- df %>% dplyr::mutate(across(-time, ~RcppRoll::roll_mean(.x, n = roll_mean_window, fill = NA, align="right")))
+        df <- df %>% tail(-(roll_mean_window - 1)) # Remove NA rows created at the beginning of df.
```

Comment:
What if the number of rows is the same as roll_mean_window?

Reply:
Added error handling.